segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -1223,9 +1226,7 @@ private CompletableFuture<SystemSnapshotRecord> createSystemSnapshotRecord(Metad
                                     }
 
                                     // Add to the system snapshot.
-                                    synchronized (systemSnapshot) {
```

Comment:
Why this is not needed anymore?

Reply:
Ok, so it was using a plain `new ArrayList<>()` before, right? Not a `Collections.synchronizedList()`

Reply:
it was still using Collections.synchronizedList(). but had additional redundant guard which is really unnecessary